### PR TITLE
6703: changes in preparation for switching specs from IAM to SIS

### PR DESCRIPTION
**Pull request by kpethtel**
## Summary

Fixes small issues that were encountered while working on the IAM to SIS spec change. 

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/6703

## Testing done
Specs.

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
Shouldn't impact anything. 

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/mobile/app/controllers/mobile/v0/check_in_controller.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def create
         if (error = parse_error(attributes))
           raise(*error)
         else
-          render json: Mobile::V0::CheckInSerializer.new(@current_user.id, attributes)
```
Comment (kpethtel) [inline modules/mobile/app/controllers/mobile/v0/check_in_controller.rb:15]: This was one of the issues that prompted the SIS spec switch. This method only exists on IAM users.

File: modules/mobile/app/controllers/mobile/v0/facilities_info_controller.rb [hunk 2]
```diff
@@ -20,7 +20,8 @@ def index
       end
 
       def schedulable
-        facility_ids = @current_user.va_treatment_facility_ids + @current_user.cerner_facility_ids
+        facility_ids = (@current_user.va_treatment_facility_ids + @current_user.cerner_facility_ids).uniq
```
Comment (kpethtel) [inline modules/mobile/app/controllers/mobile/v0/facilities_info_controller.rb:23]: I believe the old specs just don't have any duplicates but the new specs do and revealed this issue.

File: modules/mobile/spec/support/schemas/v2/user.json [hunk 3]
```diff
@@ -38,7 +38,7 @@
               "type": "string"
             },
             "middleName": {
-              "type": "string"
+              "type": ["string", "null"]
```
Comment (kpethtel) [inline modules/mobile/spec/support/schemas/v2/user.json:41]: This came up because the default test user doesn't have a middle name.

**Review by aherzberg — APPROVED** [review]
Nice catches. Glad we're doing this switch to SIS in our specs

**Pull request merged**